src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/ReflectionInvokeMapNode.cs

```diff
@@ -99,12 +99,25 @@ internal static void AddSignatureDependency(ref DependencyList dependencies, Nod
             {
                 factory.TypeSystemContext.DetectGenericCycles(type, referent);
 
+                // If the user can legitimately get to a System.Type instance by inspecting the signature of the method,
+                // the user then might use these System.Type instances with APIs such as:
+                //
+                // * MethodInfo.CreateDelegate
+                // * RuntimeHelpers.Box
+                // * Array.CreateInstanceForArrayType
+                //
+                // All of these APIs are trim/AOT safe and allow creating new instances of the supplied Type.
+                // We need this to work.
+                // NOTE: We don't have to worry about RuntimeHelpers.GetUninitializedObject because that one is annotated
+                // as trim-unsafe.
```

Comment:
Would it be worth it to make these cases conditional on the corresponding APIs having been used?

Reply:
I'm not convinced this is the full list of APIs (it's not full list for delegates) and it's not worth adding expansive infrastructure for what is at best a 0.7% saving per rt-sz.

Reply:
Makes sense to me.